hazelcast-all/src/test/java/com/hazelcast/it/DependencyReducedPomIT.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.it;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.FileInputStream;
+import java.util.Iterator;
+
+import javax.xml.namespace.NamespaceContext;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathFactory;
+
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.w3c.dom.Document;
+import org.w3c.dom.NodeList;
+
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.QuickTest;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category(QuickTest.class)
+public class DependencyReducedPomIT {
+
+    @Test
+    public void testZeroCompileScopedDeps() throws Exception {
```

Comment:
@kwart @jerrinot I think we need to add similar fail safe to release automation, I will work on it, please let me know if you see that redundant after this test case.

Reply:
@kwart I also see below when we execute mvn deploy

Reply:
> I think we need to add similar fail safe to release automation, I will work on it, please let me know if you see that redundant after this test case.

We should ask @devozerov for an explanation. I think the 2 different shaded artifacts are intentional, but I didn't dig into it to get the reasoning.

Reply:
In the SQL module we have two executions of the `shade` module:
1. `fat-shaded-jar` - the default Jar with all dependencies merged into the `hazelcast-sql.jar`, this is what we publish 
1. `slim-original-jar` (previously `fat-jar`) produces a JAR that is convenient for the Jet team development process, it is not published (@gierlachg may provide additional details, my explanation could be inaccurate)

Reply:
Hmm, why it is not published? We rely on it.